src/main/java/uk/gov/pay/api/model/PaymentState.java

```diff
@@ -71,4 +77,21 @@ public String toString() {
                 ", code=" + code +
                 '}';
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
might it not be easier to use https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/builder/EqualsBuilder.html?